docs/windows.md

```diff
@@ -1,142 +1,94 @@
 # Building on Windows
 
-## Install Git For Windows
-
-As Git is used by the Go language to download dependencies, you need
-to install Git, for example from http://git-scm.com/.
-
-You also must make sure that the directory that contains the Git For
-Windows binary is in the Path environment variable. Note that Git For
-Windows has a 'git' binary in a 'Git\bin' directory and another one in
-a 'Git\cmd' directory. You should only put the 'Git\cmd' directory in
-the Path environment variable.
-
 ## Install Go
 
-Please install the Go language as explained on
-https://golang.org/doc/install.
-
-To properly install Go, you will need to set some environment
-variables. We recommend you to set them globally using the Control
-Panel, as explained in the documentation above, so that these
-environment variables are automatically available in all the possible
-environments that you might want to use like Git Bash, Windows's cmd,
-Cygwin's terminal, Windows' PowerShell and so on.
-
-You must make sure that the GOROOT environment variable is set and
-that the %GOROOT%/bin directory is in the Path environment variable.
-
-The GOPATH environment variable should also be set to a directory that
-you have created, and the %GOPATH/bin directory should also be in the
-Path environment variable.
-
-## Download go-ipfs and fix Git authentication
+`go-ipfs` is built on Golang and thus depends on it for all building methods.  
+https://golang.org/doc/install  
+The `GOPATH` environment variable must be set as well.  
+https://golang.org/doc/code.html#GOPATH
 
-Use the following command to download go-ipfs source code:
-
-```
+## Choose the way you want to proceed
+`go-ipfs` utilizes `make` to automate builds and testing, but can be built without it using only `git` and `go`.  
+The `make` portions cover setting up the environment, "minimal" assumes you have a working version of `go` and `git` already.  
+[make→](#make)  
+[minimal→](#minimal)
+
+## Make
+There are currently 2 documented ways of building with `make` on Windows.  
+[MSYS2→](#msys2)  
+[Cygwin→](#cygwin)
+
+## MSYS2
+Install msys2 (http://www.msys2.org)  
+Run the following inside a normal `cmd` prompt (Not the MSYS2 prompt, we only need MSYS2's tools).  
+An explanation is below.
+```
+SET PATH=%PATH%;\msys64\usr\bin
+pacman --noconfirm -S  git make unzip
 go get -u github.com/ipfs/go-ipfs
+cd %GOPATH%\src\github.com\ipfs\go-ipfs
+make install
 ```
 
-The above command uses Git to download go-ipfs from its GitHub
-repository. If you get authentication problems with Git, you might
-want to take a look at
-https://help.github.com/articles/caching-your-github-password-in-git/
```

Comment:
Is this still a common issue? If so, it seems like the advice here should be kept, maybe in a trouble-shooting section at the end?

Reply:
I'm not sure the origin of this. Even years ago I never encountered this on any of the environments I used for testing. It may have been an issue upstream that was resolved.

Either way the troubleshooting section is a good idea for all misc. comments.